### ISSUE TITLE
Documentation inheritance via copydocstring

### ISSUE THREAD

**Issue by NealHumphrey** [DistrictDataLabs/yellowbrick#152]
It would be great if we didn't have to include duplicate documentation for every class that inherits from base visualizer, (e.g. fit method that defines X and y every time). 

This StackOverflow question describes a @copydoc  decorator that might be used to let classes inherit the docstrings from base classes that can still be modified by the child class: 

Other ideas for implementation: 
- use a decorator to directly inherit from parents via manual specification [as described on StackOverflow](http://stackoverflow.com/questions/13741998/is-there-a-way-to-let-classes-inherit-the-documentation-of-their-superclass-with)
- Use a metaclass that allows for docstring inheritance [as described in the Sphinx issues](https://github.com/sphinx-doc/sphinx/issues/3140)
- Make building blocks of our most frequently repeated documentation (i.e. the fit method) and define them explicitly using a [related decorator function](http://stackoverflow.com/questions/12218578/setting-the-docstring-to-an-expression-inside-def/12218693#12218693)

Investigate options for implementing this and determine if any of them will work for Yellowbrick

**Comment by NealHumphrey** [comment]
@rebeccabilbro I found a few solutions to the docstring issue. Might be worth trying to tackle. From a brief read the metaclass looks the cleanest. @bbengfort, wondering if you've looked at and decided against any of these options previously, sounded like from Rebecca you all have discussed and grudgingly accepted the repeated-docstring version.

**Comment by NealHumphrey** [comment]
Another example
http://stackoverflow.com/questions/2025562/inherit-docstrings-in-python-class-inheritance

**Comment by NealHumphrey** [comment]
@bbengfort  I found this in `.utils` - looks like you added it back in November 2016? Did you ever test out using it? 

https://github.com/DistrictDataLabs/yellowbrick/blob/master/yellowbrick/utils.py#L189

**Comment by bbengfort** [comment]
@NealHumphrey yes - I was going to mention that when I saw this issue, but it got lost I guess. So I did try to create the decorator, the problem was that it was causing problems with Sphinx and I was having a hard time getting the string concatenation to work correctly. You're right, there ends up being a lot of documentation repetition, so if you want to dive further into this, I'd be happy to check it out with you. 

Before we get too far down this road, however, I'd say that the docstrings do need an overhaul. We had several different styles of docstrings used by different contributors, and not all of them are standard.
